packages/sidecar/src/messageBuffer.ts

```diff
@@ -68,10 +68,16 @@ export class MessageBuffer<T> {
   }
 
   read(): T[] {
```

Comment:
This does something _very_ similar to `.stream()` method above so we should align them or make them share some code. Also it might be wise to make `start` an argument but defaulting to `this.head` for different clients being at different places?